internal/engine/compiler/engine.go

```diff
@@ -880,6 +880,9 @@ func (f *function) getSourceOffsetInWasmBinary(pc uint64) uint64 {
 		}
 		return srcMap.irOperationOffsetsInNativeBinary[i] >= pcOffsetInNativeBinary
 	})
+	if index == 0 && len(srcMap.irOperationSourceOffsetsInWasmBinary) > 0 {
```

Comment:
I think I need comment on this for the future

Reply:
personally I would ask "why is this done (only) on index zero" and want the comment to answer that. Up to others what the good comment would be

Reply:
I wonder if it's worth adding a `func hasSourceOffsetsInWasmBinary() bool { return len(irOperationSourceOffsetsInWasmBinary) != 0 }` (maybe not in this PR)

Reply:
Added a comment to clarify.